### PR TITLE
Make absinthe_plug an optional runtime dependency

### PR DESCRIPTION
**Pull request by lytedev**
Per:
https://github.com/opencensus-beam/opencensus_absinthe/pull/13/files#r290670529

This resolves a race condition at compile-time. See upstream PR here:
https://github.com/opencensus-beam/opencensus_absinthe/pull/13

### CODE REVIEW

**Comment by lytedev** [comment]
I went ahead and lumped in my changes for #15 into this PR. If it makes anybody upset, I can rebase/cherry pick as necessary.

**Comment by garthk** [comment]
I'd like to keep these separate, yes.

**Comment by garthk** [comment]
I've split the race condition from PR #13 into PR #16. @lytedev, please:

* `git checkout -b` a new branch name that reflects #15
* rebase and drop `cb585de`
* file a new PR with a title that reflects #15

File: lib/opencensus/absinthe/plug.ex [hunk 1]
```diff
@@ -28,8 +28,8 @@ defmodule Opencensus.Absinthe.Plug do
     """
     def traced_pipeline(config, pipeline_opts \\ []) do
       config
-      |> Absinthe.Plug.default_pipeline(pipeline_opts)
       |> Opencensus.Absinthe.add_phases()
+      |> Absinthe.Plug.default_pipeline(pipeline_opts)
```
Comment (garthk) [inline lib/opencensus/absinthe/plug.ex:32]: Did this work? If so, the Absinthe documentation might need some work. The documentation for [`Absinthe.Plug/default_pipeline/2`](https://hexdocs.pm/absinthe_plug/Absinthe.Plug.html#default_pipeline/2) says it takes a map and a keyword list and returns a pipeline, i.e. a list of phases…

Comment (lytedev) [reply lib/opencensus/absinthe/plug.ex:32]: > Did this work? If so, the Absinthe documentation might need some work. The documentation for [`Absinthe.Plug/default_pipeline/2`](https://hexdocs.pm/absinthe_plug/Absinthe.Plug.html#default_pipeline/2) says it takes a map and a keyword list and returns a pipeline, i.e. a list of phases…

Probably not. As you say, this looks like I have lines 31 and 32 in the wrong order. 🤦‍♂ 

Nice catch!

**Pull request closed without merging**